src/routes/userRoutes.js

```diff
@@ -0,0 +1,42 @@
+const { Router } = require('express');
+const {
+  GetAllUsersController,
+  GetUserController,
+  CreateUserController,
+  EditUserController,
+  DeleteUserController,
+} = require('../controllers/user');
+const {
+  verifyIfIdIsANumber,
+  verifyIfUserExists,
+  verifyUserFields,
+} = require('../middlewares');
+
+const userRoutes = Router();
+
+userRoutes.get('/', GetAllUsersController.handle);
+
+userRoutes.get(
+  '/:id',
+  verifyIfIdIsANumber,
+  verifyIfUserExists,
+  GetUserController.handle,
+);
+userRoutes.post('/', verifyUserFields, CreateUserController.handle);
+
+userRoutes.patch(
+  '/:id',
+  verifyIfIdIsANumber,
+  verifyIfUserExists,
+  verifyUserFields,
```

Comment:
_verifyUserFields_ should come before _verifyIfUserExists_.
Why? Because if this verification fails, there is no need to access and search the DB (which is done by _verifyifUserExists_)